### ISSUE TITLE
Bug in From field when includes combination of double quotes and non-ascii characters

### ISSUE THREAD

**Issue by annika07** [mikel/mail#1135]
I found that Mail fails to send an email when the from field contains a combination of double quotes and non-ascii characters (the address ends up malformed).
Here is the example:

```ruby
require 'mail'

Mail.defaults do
    delivery_method :smtp, address: "localhost", port: 1025
end

from_addr = Mail::Address.new
from_addr.display_name = "Τεστ \" Test"
from_addr.address = "me@testmail.com"
puts "From field:  #{from_addr}"
puts

mail = Mail.new do
    from  from_addr
    puts "From address: #{from}"
    puts
    to  "you@testmail.com"
    subject "Test mail"
end

puts mail
mail.deliver
```

The from field is a combination of Greek, double quotes and English and the resulted message is the following:

```
From field:  "Τεστ \" Test" <me@testmail.com>

From address: "Τεστ \" Test" <me@testmail.com>

Date: Fri, 21 Jul 2017 14:54:01 -0700
From: =?UTF-8?Q?=22=CE=A4=CE=B5=CF=83=CF=84_\=22_Test=22_<me@testmail.com>?=
To: you@testmail.com
Message-ID: <597277f9ef6f9_6fd53feff0c5e1ec85494@C02SR275H03M.mail>
Subject: Test mail
Mime-Version: 1.0
Content-Type: text/plain
Content-Transfer-Encoding: 7bit
```

As you see, when the mail is created, the from address is not in the correct format ["me@testmail.com"].

If you replace the Greek characters with English, and make the from field as \"Test \\" Test\", the resulted message is 

```
From field:  "Test \" Test" <me@testmail.com>

From address: ["me@testmail.com"]

Date: Fri, 21 Jul 2017 15:00:40 -0700
From: "Test \" Test" <me@testmail.com>
To: you@testmail.com
Message-ID: <5972798890fdc_702f3ff42345e1f4612cf@C02SR275H03M.mail>
Subject: Test mail
Mime-Version: 1.0
Content-Type: text/plain
Content-Transfer-Encoding: 7bit
```

It seems that the list of addresses fails to be created, and my assumption is that something is wrong in the FieldList class in /mail/lib/mail/field_list.rb, because until that point the mail field is created correctly.

**Comment by jeremy** [comment]
Thanks for the report, @annika07! This is working on Mail 2.7.0.rc1. Fixed by #1103.

**Comment by annika07** [comment]
Hi @jeremy 

I run the above script using Mail 2.7.0.rc1 and it does not work. The mail is never sent and the address is still malformed (From: =?UTF-8?Q?=22=CE=A4=CE=B5=CF=83=CF=84_\=22_Test=22_<me@testmail.com>?=).

If I use "puts Gem.loaded_specs["mail"].version" it prints 2.7.0.rc1, so I am definitely using that. 
Are you able to verify that the above email is sent using Mail 2.7.0.rc1?
